src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -2234,6 +2246,10 @@ public void execute() throws GitException, InterruptedException {
                     listener.getLogger().println("[ERROR] JGit doesn't support submodule update --reference yet.");
                     throw new UnsupportedOperationException("not implemented yet");
                 }
+                if (shallow) {
+                    listener.getLogger().println("[ERROR] JGit doesn't support shallow submodules yet.");
+                    throw new UnsupportedOperationException("not implemented yet");
```

Comment:
Hi @MarkEWaite, should using `shallow` for JGit throw this `UnsupportedOperationException` here or just log a WARNING when calling `shallow(true)` at lines 2224-2227?

As a reference `FetchCommand` and `CloneCommand` just perform logging. That's not consistent in this class.

Reply:
You're right @darxriggs , it would be good enough to log the implementation gap rather than fail the checkout.  Shallow clone is a performance optimization that isn't critical to the typical use of the repository.